src/Atmos/Model/bc_momentum.jl

```diff
@@ -0,0 +1,76 @@
+abstract type MomentumBC end
+
+"""
+    Impenetrable(drag::MomentumDragBC) :: MomentumBC
+
+Defines an impenetrable wall model for momentum.
+"""
+struct Impenetrable{D} <: MomentumBC
+  drag::D
+end
+
+function atmos_boundary_state!(nf::NumericalFluxNonDiffusive, bc_momentum::Impenetrable, atmos,
+    state⁺, aux⁺, n, state⁻, aux⁻, bctype, t, args...)
+  state⁺.ρu -= 2*dot(state⁻.ρu, n) .* SVector(n)
```

Comment:
Can we move this to `atmos_boundary_state!(nf::NumericalFluxNonDiffusive, bc_momentum::Impenetrable{FreeSlip}, args...)` and then have `atmos_boundary_state!(nf::NumericalFluxNonDiffusive, bc_momentum::Impenetrable{DrawLaw}, args...)` call that function? 

That way we aren't setting a state and then immediately overwriting it for the `NoSlip` case.